### ISSUE TITLE
Greater than one dimensional fitting

### ISSUE THREAD

**Issue by JCCPort** [go-hep/hep#707]
It would be great to have the ability to fit to a function where there is more than 1 independent variable changing.

```
res, err := fit.CurveND(
    fit.FuncND{
        F: func(x []float64, ps []float64) float64 {
            return testFunction(x, ps[0], ps[1])
        },
        X:  xdata,
        Y:  ydata,
        N:  2,
    },
    nil, &optimize.NelderMead{},
)

**Comment by sbinet** [comment]
sounds good to me.
do you want to try sending a PR?

**Comment by JCCPort** [comment]
I can give it a crack, might be a while as I'm still getting to grips with GoLang

**Comment by sbinet** [comment]
Fixed #708.
